### ISSUE TITLE
Captions duplicated and too big in Vimeo

### ISSUE THREAD

**Issue by matoni555** [sampotts/plyr#877]
Hello,
I use player with Vimeo video with captions. But captions are duplicated and too big:
![plyr-vimeo-captions](https://user-images.githubusercontent.com/3002995/38487928-19048d5c-3be2-11e8-9404-de842163657d.png)

### Expected behaviour
Not duplicated captions. Too big captions hidden.

### Actual behaviour
Captions duplicated and too big.

### Environment

- Browser: Chrome
- Version: 65.0.3325.181
- Operating System: Windows
- Version: 11

Thank you

**Comment by sampotts** [comment]
What size screen? I'm not sure we can do a lot about this as Vimeo doesn't let you disable their caption rendering via their API.

**Comment by matoni555** [comment]
I use iframe size same as in screenshot above: 338x190 px.
It is strange that in Vimeo captions example are captions small: https://help.vimeo.com/hc/en-us/articles/224968828-Captions-and-subtitles

**Comment by sampotts** [comment]
This should be resolved in `v3.2.0` 👍

**Comment by matoni555** [comment]
@sampotts Captions not works in Vimeo video player now (v3.3.7). Check this demo: http://www.matonoha.cz/_html/plyr/

**Comment by friday** [comment]
I think that size combined with multi-line cues is more than the css hack can handle, but maybe Sam has some tricks up his sleeves.

If you meant the actual size, it is ok imo. I wouldn't want it much smaller anyway. If you do you can change it with css. Perhaps it could be positioned closer to the bottom with small iframes/viewports.

**Comment by sampotts** [comment]
This is a tricky one. I need to take another look as Vimeo doesn't seem to allow you to turn off their rendering of the captions without them also not being available via the API if I remember correctly. The offset hack to hide the controls also means the captions get larger (they must use some sort of responsive type mechanism) so it's a fine balance. I'll take another look as this is a good example to test with.

**Comment by samhibberd** [comment]
Do we think this is something that can be fixed? If not has anyone else discovered any other work arounds? Thanks, Sam

**Comment by friday** [comment]
It's not possible to get cues via the Vimeo API unless you enable captions. If you enable them it's not possible to hide them *safely*. The css hack which is currently used doesn't work for all resolutions.

So no.

You can disable captions or not use Vimeo.

**Comment by samhibberd** [comment]
Thanks for the options @friday

**Comment by xnetdude** [comment]
How do you disable captions if you don't have control over the source vimeo material?

**Comment by diegoalex** [comment]
Hi Guys, I came up to the same issue here.
I've been trying to do some workaround but with no success.
My first idea was to hide the caption inside the iFrame with javascript, but I get cross-origin issue .

The easier way to 'fix' it would be to force a display:none in the class '.vp-captions' inside the iframe. But i cant do that =(
![image](https://user-images.githubusercontent.com/3237570/86941004-cb74e400-c119-11ea-9396-1ab8b17b3914.png)

Any of you guys found a way to fix it?

Thanks

**Comment by friday** [comment]
If you want to control aspects like this hosting your own HTML5 video or streaming it is the only way. Vimeo and YouTube will always want to put their own brand on the player, which is reasonable seeing as they pay for the hosting and bandwith.

**Comment by diegoalex** [comment]
> If you want to control aspects like this hosting your own HTML5 video or streaming it is the only way. Vimeo and YouTube will always want to put their own brand on the player, which is reasonable seeing as they pay for the hosting and bandwith.

But the Subtitle issue is something related to PLYR, when I open the vimeo video embed somewhere it does't happen.
If you check the CSS inside the iFrame you can see that:
![image](https://user-images.githubusercontent.com/3237570/86947683-7e494000-c122-11ea-98ce-fc3b85ec8cd3.png)

For some reason, when you're wrapped by the PLYR it adds this huge font-size.

**Comment by explorier** [comment]
Just wanted to bump this issue, encountered it after thinking Vimeo captions were working well. The strange thing is they worked fine when testing on this video embed:
`      <iframe
        src="https://player.vimeo.com/video/76979871?loop=false&amp;byline=false&amp;portrait=false&amp;title=false&amp;speed=true&amp;transparent=0&amp;gesture=media"
        allowfullscreen
        allowtransparency
        allow="autoplay"
      ></iframe>`

But then when I actually plugged in the client's videos and subtitles, the huge captions appeared. I'm not sure exactly what the difference is here, maybe just the example video had short enough captions that the huge captions didn't quite break out of the hidden iframe?

**Comment by sjonkeesse** [comment]
I'm experiencing the same issue as well. I've noticed that it only happens with long sentences. When I reduce the .vp-captions font-size by ~5%, they don't seem to appear anymore (with my sentence lengths). Is there a way that I can tune the 'offset hack'? That way I can potentially reduce the duplicate subtitles somewhat.

**Comment by michelstomp** [comment]
> I'm experiencing the same issue as well. I've noticed that it only happens with long sentences. When I reduce the .vp-captions font-size by ~5%, they don't seem to appear anymore (with my sentence lengths). Is there a way that I can tune the 'offset hack'? That way I can potentially reduce the duplicate subtitles somewhat.

I see this working as well when I insert it straight into my browser inspector. But not when adding it to my CSS file, or as inline style in my DOM. It does not affect elements in the iframe. How should this work?

**Comment by altechzilla** [comment]
I'm getting the same issue on Vimeo-hosted videos passed through Plyr (latest version on Chrome macOS). Looks like it's only happening when captions are a bit longer. 

Here's a demo:
[Demo Screen Recording](https://share.getcloudapp.com/E0u49gAY)

Any thoughts on what can be done @sampotts ?

Thank you!

**Comment by altechzilla** [comment]
Following up here, @sampotts - my devs couldn't find a quick workaround. Any idea on why Plyr might mess up formatting for longer Vimeo captions?

**Comment by sampotts** [comment]
It's because we can't actually hide the Vimeo captions. I'd need to check again but there was previously no way to disable the built-in captions via the available options.

**Comment by altechzilla** [comment]
Yeah, but it doesn’t happen with the regular Vimeo captions and it’s only for captions that are longer.

Are you sure it’s not something related to the Plyr wrapper and the way it affects the iFrame?

**Comment by sampotts** [comment]
You can try it. Get the iframe URL, open it and resize the window to full screen. Big captions. Because we only resize vertically and that must be what Vimeo bases the resizing on, it makes the problem worse as the captions wrap and become taller vertically, overlapping into the part of the player visible. The only ways it can be resolved are:

- we disable the custom caption rendering
- Vimeo adds an option to their player.js API

**Comment by Tumbuctoo** [comment]
Could we ALL try to communicate with Vimeo so that they add the proper option to their player.js.API?
How could we all of us, as a bunch, make sure they know we would need them to add such an option?
Sorry if I seem too naive but things also work like this (sometimes); by talking. Any ideas?

**Comment by altechzilla** [comment]
Did anybody find a workaround?

**Comment by Tumbuctoo** [comment]
As said, it seems as there is no workaround unless talking to Vimeo staff.

**Comment by crispycrispycrispy** [comment]
Hey guys 
I've been at this issue for some time now, since I use it for production

I have found a fix but need to test further
It actually uses an open issue in Vimeo's player.js [here](https://github.com/vimeo/player.js/issues/501)
I actually initialize the vimeo player with `background: true` which inadvertently also removes the exaggerated captions...

It sounds so hacky but it is a quick fix.

I love this player and am open to feedback and tests about this fix.
I can also create a PR if necessary

Thanks guys for your efforts.

**Comment by darkylmnx** [comment]
Same issue happening here. Got this in production...

**Comment by irasantiago** [comment]
i just opened this related issue. it might help us all get it on Vimeo's radar if there are more thumbs up on it https://github.com/vimeo/player.js/issues/747

**Comment by reubenfield** [comment]
> Hey guys I've been at this issue for some time now, since I use it in production.
> 
> I have found a fix but need to test further... It actually uses an open issue in Vimeo's player.js [here](https://github.com/vimeo/player.js/issues/501) I actually initialize the vimeo player with `background: true` which inadvertently also removes the exaggerated captions...
> 
> It sounds so hacky but it is a quick fix.
> 
> I love this player and am open to feedback and tests about this fix. I can also create a PR if necessary
> 
> Thanks guys for your efforts.

Hi @crispycrispycrispy and @modi98. Is this still working for you in early 2024? Would you mind please sharing a link?

**Comment by alevals** [comment]
Any update on this?
This issue is critical for us, without a fix we'll need to find a different video player or switch to the original Vimeo player

**Comment by matthieua** [comment]
> Any update on this? This issue is critical for us, without a fix we'll need to find a different video player or switch to the original Vimeo player

Yes, same here. This is creating a lot of issues with our users.

**Comment by oalexdoda** [comment]
*cries in 2018* - any other libraries out there you guys are exploring?

**Comment by sampotts** [comment]
I really wished this was an easy fix or I'd jump on and sort it but the truth is it's always been a bit of a cat and mouse game with Vimeo and (to an extent) YouTube with me working around their native controls. Vimeo seem to have cottoned onto my trick of making the iframe super tall and then clipping it so we only get the right ratio. I'll try and take a quick look and see if I can come up with anything. I suspect it's a problem that's going to be the case for any library supporting Vimeo videos though, sadly. There was no way to disable the built-in captions when I last looked. 

Failing that, if you wanted to check out [Vidstack](https://www.vidstack.io/), it's a more modern alternative. Eventually, I'll be able to turn my attention there and help @mihar-22 carry on the awesome work he's been doing on it.

**Comment by sampotts** [comment]
I'll try the `cc` parameter on the frame tonight and see if that fixes it 🤞🏼

**Comment by sampotts** [comment]
I did some digging and tried a few different options but whenever `enableTextTrack` is called, the default (Vimeo's) captions are showing, hence the double up. I can't see an obvious way to hide _their_ captions while still getting the `cuechange` events etc. I'll open an issue. 

On a positive note, they seem to have opened up the `controls` option to non pro/premium accounts so I'll be able to rip out a load of ropey logic to stretch the iframe to hide their controls.

**Comment by sampotts** [comment]
I've created an issue on Vimeo's SDK:
https://github.com/vimeo/player.js/issues/1049

**Comment by realanthonyc** [comment]
hi. I am trying to embed https://vimeo.com/255546259, but the Caption is still duplicated.

For now, hide Plyr's caption by CSS:
`
.plyr--vimeo .plyr__captions,
.plyr--vimeo [data-plyr="captions"] {
  display: none;
}
`

**Comment by jmc-emperor** [comment]
+1, same issue here, would love if a fix could be implemented

**Comment by sampotts** [comment]
I'll work on getting this fixed up.

**Comment by xini** [comment]
Thanks! 🎉
